### PR TITLE
Data Collection Grants

### PR DESCRIPTION
**Pull request by pettinarip**
## Description

Adds the new Data Collection Grants application.

Preview: https://deploy-preview-313--ecosystem-support.netlify.app/data-collection-grants

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -173,6 +173,21 @@ export const SIDEBAR_RUN_A_NODE__GRANTS_LINKS: SidebarLink[] = [
   }
 ];
 
+export const DATA_COLLECTION_ROUND_URL = '/data-collection-grants';
+export const SIDEBAR_DATA_COLLECTION_GRANTS_LINKS: SidebarLink[] = [
+  { text: 'Introduction', href: `${DATA_COLLECTION_ROUND_URL}/#introduction` },
+  { text: 'Ideas', href: `${DATA_COLLECTION_ROUND_URL}/#ideas` },
+  { text: 'Submit A Proposal', href: `${DATA_COLLECTION_ROUND_URL}/#submit-proposal` },
+  { text: 'Eligibility Criteria', href: `${DATA_COLLECTION_ROUND_URL}/#eligibility-criteria` },
+  { text: 'Eligible', href: `${DATA_COLLECTION_ROUND_URL}/#eligible` },
```
Comment (wackerow) [inline src/constants.ts:182]:
```suggestion
  { text: 'Ineligible', href: `${DATA_COLLECTION_ROUND_URL}/#ineligible` },
```

File: src/pages/data-collection-grants/index.tsx [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+                    Projects at any point in the development process: just an idea, early stages,
+                    proof of concept, or with significant progress already made. However, we do not
+                    fund past work
+                  </ListItem>
+                  <ListItem>Builders of any age, origin, identity or background</ListItem>
+                </List>
+              </section>
+
+              <section id='eligible' ref={ref6}>
+                <PageSection mb={6}>Eligible</PageSection>
```
Comment (wackerow) [inline src/pages/data-collection-grants/index.tsx:173]:
```suggestion
              <section id='ineligible' ref={ref6}>
                <PageSection mb={6}>Ineligible</PageSection>
```

**Review by wackerow — APPROVED** [review]
Nice job @pettinarip, looks good!

**Review by wackerow — APPROVED** [review]

File: src/components/layout/DataCollectionLayout.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+          alt: 'Two people holding hands in field with ethereum log',
+          src: mergeDataChallengeHeroMobile
+        }}
+        title='Data Collection Grants 2023'
+      >
+        <Text>
+          A proactive grants round to improve the community&apos;s data collection capabilities.
+        </Text>
+
+        <Text>Proposals are due at 11:59 UTC October 23rd 2023</Text>
```
Comment (corwintines) [inline src/components/layout/DataCollectionLayout.tsx:28]:
```suggestion
        <Text>Proposals are due at 11:59 UTC October 23rd 2023.</Text>
```

File: src/pages/data-collection-grants/index.tsx [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+            <Stack spacing={10}>
+              <section id='introduction' ref={ref2}>
+                <PageSection mb={6}>Introduction</PageSection>
+
+                <PageText mb={6}>
+                  The Ecosystem Support Program is launching this grants initiative to encourage
+                  research and development around data collection and visualization of the Ethereum
+                  network. We are looking for builders to work on testnets, data collection and
+                  XATU. The Ethereum ecosystem needs more free and open source infrastructure for
+                  researchers, developers, and data analysts and developers—read below and apply!
```
Comment (corwintines) [inline src/pages/data-collection-grants/index.tsx:68]:
```suggestion
                  researchers, developers, data analysts and developers—read below and apply!
```

One too many ands here

File: src/components/forms/DataCollectionForm.tsx [hunk 5]
```diff
@@ -0,0 +1,575 @@
...
+
+          <Flex direction={{ base: 'column', md: 'row' }} gap={12} mb={3}>
+            <Controller
+              name='country'
+              control={control}
+              render={({ field: { onChange }, fieldState: { error } }) => (
+                <Field
+                  id='country'
+                  label='Country'
+                  helpText='Provide the country of where the node would be located'
```
Comment (corwintines) [inline src/components/forms/DataCollectionForm.tsx:175]: I think this copy is from the run a node round

File: src/components/Banners.tsx [hunk 6]
```diff
@@ -1,7 +1,29 @@
 import { FC } from 'react';
+import { useRouter } from 'next/router';
+import { Box, Link } from '@chakra-ui/react';
+
+import { Banner } from './UI';
+
+import { DATA_COLLECTION_ROUND_URL } from '../constants';
 
 interface Props {}
```
Comment (nhsz) [inline src/components/Banners.tsx:9]: empty interface can be removed

File: src/components/Banners.tsx [hunk 7]
```diff
@@ -1,7 +1,29 @@
...
 
 export const Banners: FC<Props> = () => {
```
Comment (nhsz) [inline src/components/Banners.tsx:11]:
```suggestion
export const Banners: FC = () => {
```

File: src/components/forms/DataCollectionForm.tsx [hunk 8]
```diff
@@ -0,0 +1,575 @@
...
+    <Stack
+      w='100%'
+      bgGradient='linear(to-br, brand.newsletter.bgGradient.start 10%, brand.newsletter.bgGradient.end 100%)'
+      px={{ base: 5, md: 12 }}
+      pt={{ base: 8, md: 12 }}
+      pb={{ base: 20, md: 16 }}
+      borderRadius={{ md: '10px' }}
+    >
+      <FormProvider {...methods}>
+        <form id='run-a-node-grants-form' onSubmit={handleSubmit(onSubmit)} noValidate>
```
Comment (nhsz) [inline src/components/forms/DataCollectionForm.tsx:107]: form `id` seems incorrect

File: src/components/forms/fields/Field.tsx [hunk 9]
```diff
@@ -1,14 +1,14 @@
...
 
 import { PageText } from '../../UI';
 
 export interface Props extends FormControlProps {
-  children: React.ReactNode;
+  children: ReactNode;
   id: string;
   label: string;
-  helpText?: string;
+  helpText?: ReactNode;
```
Comment (nhsz) [inline src/components/forms/fields/Field.tsx:11]:
```suggestion
  helpText?: string;
```

File: src/pages/data-collection-grants/index.tsx [hunk 10]
```diff
@@ -0,0 +1,297 @@
...
+              <section id='what-is-not-eligible' ref={ref6}>
+                <PageSection mb={6}>What is NOT eligible</PageSection>
+
+                <List>
+                  <ListItem>
+                    Anything that is not legal within the jurisdiction where the work is taking
+                    place
+                  </ListItem>
+                  <ListItem>
+                    Financial products (trading, investment products, lending, betting etc)
```
Comment (nhsz) [inline src/pages/data-collection-grants/index.tsx:181]:
```suggestion
                    Financial products (trading, investment products, lending, betting, etc)
```

**Review by nhsz — COMMENTED** [review]
@pettinarip I left some comments

File: src/components/forms/constants.ts [hunk 11]
```diff
@@ -1483,6 +1492,15 @@ export const APPLYING_AS_OPTIONS = [
   { value: 'Other', label: 'Other' }
 ];
 
+export const APPLYING_AS_DATA_COLLECTION_OPTIONS = [
```
Comment (nhsz) [inline src/components/forms/constants.ts:1495]: This const should be removed and should use `APPLYING_AS_OPTIONS` instead

**Review by nhsz — CHANGES_REQUESTED** [review]

File: src/constants.ts [hunk 12]
```diff
@@ -194,6 +210,7 @@ export const ACADEMIC_GRANTS_THANK_YOU_PAGE_URL = '/academic-grants/thank-you';
 export const DEVCON_GRANTS_THANK_YOU_PAGE_URL = '/devcon-grants/thank-you';
 export const ECODEV_GRANTS_THANK_YOU_PAGE_URL = '/ecodev-grants/thank-you';
 export const RUN_A_NODE_GRANTS_THANK_YOU_PAGE_URL = '/run-a-node-grants/thank-you';
+export const DATA_COLLECTION_THANK_YOU_PAGE_URL = '/data-collection-grants/thank-you';
```
Comment (nhsz) [inline src/constants.ts:213]: this const is not being used

**Review by nhsz — CHANGES_REQUESTED** [review]

File: src/components/forms/constants.ts (hunk 11 above)
Comment (pettinarip) [reply src/components/forms/constants.ts:1495]: Nice, thanks!

File: src/components/forms/fields/Field.tsx (hunk 9 above)
Comment (pettinarip) [reply src/components/forms/fields/Field.tsx:11]: Sometimes we need to pass rich content like text with links inside.

File: src/constants.ts (hunk 12 above)
Comment (pettinarip) [reply src/constants.ts:213]: It is used in the form's submission logic.

**Review by nhsz — APPROVED** [review]

**Review by corwintines — APPROVED** [review]

**Pull request merged**